### ISSUE TITLE
Question About The role of Optical Flow

### ISSUE THREAD

**Issue by LittlefatChuan** [googleinterns/deep-stabilization#18]
Hello. idk why i'm just unable to achieve the result of  Figure 8 (b) red Line in your paper, i only got the blue curve with all loss function enabled.And I retrain the model without Optical flow input and disable the optical flow loss(i set noFlo = True in your code). Then i 
 noticed the rotation curve between with Optical Flow and the without one are very similar. Does the Optical Flow really matter?

![image](https://github.com/googleinterns/deep-stabilization/assets/51955380/69c2d0d0-5fe4-46ed-97d7-bdf1a16e83b5)
![image](https://github.com/googleinterns/deep-stabilization/assets/51955380/adf4f593-dcae-40d5-b104-e862ea9350c8)
![image](https://github.com/googleinterns/deep-stabilization/assets/51955380/4adad6bb-1bad-46dd-8c5a-8bfc2b6c4f7f)

**Comment by DEEPMINDing** [comment]
Hi, I would like to know how you changed the code to the version without optical flow, and what specific changes have been made? I only made changes to [no_flo=True], which resulted in this error：
size mismatch for rnns.0.LSTM.weight_ih: copying a param with shape torch.Size([2048, 156]) from checkpoint, the shape in current model is torch.Size([2048, 92]).

**Comment by LittlefatChuan** [comment]
> Hi, I would like to know how you changed the code to the version without optical flow, and what specific changes have been made? I only made changes to [no_flo=True], which resulted in this error： size mismatch for rnns.0.LSTM.weight_ih: copying a param with shape torch.Size([2048, 156]) from checkpoint, the shape in current model is torch.Size([2048, 92]).

hello, you should also tuning the model.py Line 78, "self.no_flo = False" set self.no_flo = True
and disable the optical loss in "train.py" and "inference.py"： "loss_step = model.loss(out, vt_1, virtual_inputs, real_inputs_step, flo_step, flo_back_step, real_projections_t, real_projections_t_1, real_postion_anchor,  follow = True, optical = True, undefine = True)", set optical = False.
then re-train the model to get a checkpoint file without optical loss.

**Comment by sabitarajbanshi** [comment]
@LittlefatChuan Hi, I have tried to run this according to the above code setup, but got error as;
![Screenshot from 2024-04-22 13-28-05](https://github.com/googleinterns/deep-stabilization/assets/28886429/349b7c53-d725-438f-80a0-57bac71178e9)
how can I run the inference for a video without the flo values?

**Comment by LittlefatChuan** [comment]
> @LittlefatChuan Hi, I have tried to run this according to the above code setup, but got error as; ![Screenshot from 2024-04-22 13-28-05](https://private-user-images.githubusercontent.com/28886429/324370694-349b7c53-d725-438f-80a0-57bac71178e9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NzI1NzMsIm5iZiI6MTcxMzc3MjI3MywicGF0aCI6Ii8yODg4NjQyOS8zMjQzNzA2OTQtMzQ5YjdjNTMtZDcyNS00MzhmLTgwYTAtNTdiYWM3MTE3OGU5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDIyVDA3NTExM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI0NTg0NTA2NDMxMzhkZjAzMzUwMGE1ZWZjZmNlMDYxMjM2NWYwZGU0MjFlY2M4MGU4NzUyMTZlMTQ0ZWU4ZWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7i-L4rOVbN3xropcV8cZ0MlvVnyy0j6502yc_NHaohI) how can I run the inference for a video without the flo values?

sorry, there is another param about optical flow in dataset loader that i forget to mention：
Line 148 in inference.py "test_loader = get_inference_data_loader(cf, data_path, no_flo = False)" --> noflo = True
Line 220 in train.py "train_loader, test_loader = get_data_loader(cf, no_flo = False)"  --> noflo = True

**Comment by sabitarajbanshi** [comment]
@LittlefatChuan have you successfully run this code if yes can we connect so that you can guide me through other dependencies issues? my email id; sabita.rajbanshi@naamii.org.np
